Client/Frontend/Login Management/BreachAlertsManager.swift

```diff
@@ -0,0 +1,149 @@
+    var isRetired: Bool
+    var isSpamList: Bool
+    var logoUrl: String?
+}
+
+/// A manager for the user's breached login information, if any.
+public class BreachAlertsManager {
+
+    //
+    // MARK: - Variables and Properties
+    //
+    var dataTask: URLSessionDataTask?
+    var endpointURL = "https://monitor.firefox.com/hibp/breaches"
+    var breaches: [BreachRecord] = []
+
+    //
+    // MARK: - Internal Methods
+    //
+    /// Loads breaches from Monitor endpoint.
+    public func loadBreaches() {
+        guard let url = URL(string: endpointURL) else {
+            return
+        }
+
+        dataTask?.cancel()
+
+        dataTask = URLSession.shared.dataTask(with: url) { data, response, error in
+            guard self.validatedHTTPResponse(response) != nil else {
+                print("loadBreaches(): invalid HTTP response")
+                return
+            }
+
+            if let error = error {
+                print("loadBreaches(): error: \(error)")
+                return
+            }
+
+            guard let data = data, !data.isEmpty else {
+                print("loadBreaches(): invalid data")
+                return
+            }
+
+            let decoder = JSONDecoder()
+
+            // .convertFromPascalCase
+            decoder.keyDecodingStrategy = .custom { keys in
+                let key = keys.last! // make sure array not empty
+
+                // string manipulation
+                let keyStr = key.stringValue
+                let pascalToCamel = keyStr.prefix(1).lowercased() + keyStr.dropFirst()
+
+                // CodingKey conformity
+                let codingKeyType = type(of: key)
+                return codingKeyType.init(stringValue: pascalToCamel)!
+
+            }
+
+            if let decoded = try? decoder.decode([BreachRecord].self, from: data) {
+                DispatchQueue.main.async {
+                    self.breaches = decoded
+                }
+            }
+        }
+
+        dataTask?.resume()
+
+    }
+
+    /// Compares a list of logins to a list of breaches and returns breached logins.
+    ///    - Parameters:
+    ///         - logins: a list of logins to compare breaches to
+    func compareToBreaches(_ logins: [Any])  {
+
+        for entry in logins {
+            let login = entry as! LoginRecord; // typecast for ease of use
+
+            for breach in breaches {
+                // host check
+                let loginHostURL = URL(string: login.hostname)
+                if loginHostURL?.baseDomain == breach.domain {
+                    print("compareToBreaches(): breach: \(breach.domain)")
+                }
+
+                // date check
+                let pwLastChanged = Date.init(timeIntervalSince1970: TimeInterval(login.timePasswordChanged))
+
+                let dateFormatter = DateFormatter()
+                dateFormatter.dateFormat = "yyyy-MM-dd"
+                let breachDate = dateFormatter.date(from: breach.breachDate)!
```

Comment:
Another force unwrapping mention, we shouldn't have any of that in the final version of this PR